plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryTypeMapping.java

```diff
@@ -335,6 +335,81 @@ public Object[][] bigqueryUnsupportedBigNumericTypeProvider()
+                {"NUMERIC(29, 0)", "CAST('-27182818284590452353602874713' AS DECIMAL(29, 0))"},
+                {"NUMERIC(30, 5)", "CAST(3141592653589793238462643.38327 AS DECIMAL(30, 5))"},
+                {"NUMERIC(30, 5)", "CAST(-3141592653589793238462643.38327 AS DECIMAL(30, 5))"},
+                {"NUMERIC(38, 9)", "CAST(100000000020000000001234567.123456789 AS DECIMAL(38, 9))"},
+                {"NUMERIC(38, 9)", "CAST(-100000000020000000001234567.123456789 AS DECIMAL(38, 9))"},
+                {"NUMERIC(10, 3)", "CAST(NULL AS DECIMAL(10, 3))"},
+                {"NUMERIC(38, 9)", "CAST(NULL AS DECIMAL(38, 9))"},
+        };
+    }
+
+    @Test(dataProvider = "bigqueryBigNumericTypeProvider")
+    public void testBigNumericWriteMapping(String type, String value)
+    {
+        try (TestTable table = new TestTable(getBigQuerySqlExecutor(), "test.numeric", format("(num_col %s)", type))) {
+            assertUpdate(format("INSERT INTO %s VALUES (%s)", table.getName(), value), 1);
+            assertQuery("SELECT * FROM " + table.getName(), format("VALUES (%s)", value));
+        }
+    }
+
+    @DataProvider
+    public Object[][] bigqueryBigNumericTypeProvider()
+    {
+        return new Object[][] {
+                {"BIGNUMERIC(3, 0)", "CAST(193 AS DECIMAL(3, 0))"},
+                {"BIGNUMERIC(3, 0)", "CAST(19 AS DECIMAL(3, 0))"},
+                {"BIGNUMERIC(3, 0)", "CAST(-193 AS DECIMAL(3, 0))"},
+                {"BIGNUMERIC(3, 1)", "CAST(10.0 AS DECIMAL(3, 1))"},
+                {"BIGNUMERIC(3, 1)", "CAST(10.1 AS DECIMAL(3, 1))"},
+                {"BIGNUMERIC(3, 1)", "CAST(-10.1 AS DECIMAL(3, 1))"},
+                {"BIGNUMERIC(4, 2)", "CAST(2 AS DECIMAL(4, 2))"},
+                {"BIGNUMERIC(4, 2)", "CAST(2.3 AS DECIMAL(4, 2))"},
+                {"BIGNUMERIC(24, 2)", "CAST(2 AS DECIMAL(24, 2))"},
+                {"BIGNUMERIC(24, 2)", "CAST(2.3 AS DECIMAL(24, 2))"},
+                {"BIGNUMERIC(24, 2)", "CAST(123456789.3 AS DECIMAL(24, 2))"},
+                {"BIGNUMERIC(24, 4)", "CAST(12345678901234567890.31 AS DECIMAL(24, 4))"},
+                {"BIGNUMERIC(29, 0)", "CAST('27182818284590452353602874713' AS DECIMAL(29, 0))"},
+                {"BIGNUMERIC(29, 0)", "CAST('-27182818284590452353602874713' AS DECIMAL(29, 0))"},
+                {"BIGNUMERIC(30, 5)", "CAST(3141592653589793238462643.38327 AS DECIMAL(30, 5))"},
+                {"BIGNUMERIC(30, 5)", "CAST(-3141592653589793238462643.38327 AS DECIMAL(30, 5))"},
+                {"BIGNUMERIC(38, 9)", "CAST(100000000020000000001234567.123456789 AS DECIMAL(38, 9))"},
+                {"BIGNUMERIC(38, 9)", "CAST(-100000000020000000001234567.123456789 AS DECIMAL(38, 9))"},
+                {"BIGNUMERIC(1)", "CAST(1 AS DECIMAL(1, 0))"},
+                {"BIGNUMERIC(1)", "CAST(-1 AS DECIMAL(1, 0))"},
+                // looks like Max precision is 29 when the scale is 0 in BigQuery, the same as for Numeric type,
```

Comment:
"looks like"

It's known behavior. You can see a code comment in `testNumericMapping` method.

Reply:
Looks like remved